### PR TITLE
umapinfo: add U_CheckField function, reformatting

### PR DESCRIPTION
**Pull request by rfomin**
Consistent check for "clear" with `strcmp`, slight reformatting.

### CODE REVIEW

File: Source/m_menu.c [hunk 1]
```diff
@@ -2586,15 +2586,15 @@ int G_GotoNextLevel(int *e, int *m)
 
 	if (gamemapinfo != NULL)
 	{
-		const char *n = NULL;
-		if (gamemapinfo->nextsecret[0]) n = gamemapinfo->nextsecret;
-		else if (gamemapinfo->nextmap[0]) n = gamemapinfo->nextmap;
-		else if (gamemapinfo->endpic[0] && gamemapinfo->endpic[0] != '-')
+		const char *next = NULL;
+		if (gamemapinfo->nextsecret[0]) next = gamemapinfo->nextsecret;
+		else if (gamemapinfo->nextmap[0]) next = gamemapinfo->nextmap;
```
[the remark this answers is not in the document]
Comment (rfomin) [reply Source/m_menu.c:2591]: I want to reformat this function and remove 8-space tabs. Do you mind?

[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply Source/m_menu.c:2591]: Please proceed. 👍

File: Source/u_mapinfo.h [hunk 2]
```diff
@@ -66,7 +66,8 @@ extern umapinfo_t default_mapinfo;
 extern boolean EpiCustom;
 mapentry_t *G_LookupMapinfo(int episode, int map);
 
-int U_ParseMapInfo(boolean is_default, const char *buffer, size_t length);
-void U_FreeMapInfo();
+#define check_field(s) (s && s[0] && strcmp(s, "-"))
```
Comment (fabiangreffrath) [inline Source/u_mapinfo.h:69]: Should we somehow "namespace" this macro, e.g. `u_check_field()`?

Comment (rfomin) [reply Source/u_mapinfo.h:69]: Maybe we should turn it into a function, like this:
```c
boolean U_CheckField(const char *str)
{
  return str && str[0] && strcmp(str, "-");
}
```

Comment (fabiangreffrath) [reply Source/u_mapinfo.h:69]: Right, that's nicer!

File: Source/g_game.c [hunk 3]
```diff
@@ -2427,21 +2432,17 @@ void G_WorldDone(void)
 
   if (gamemapinfo)
   {
-    if (gamemapinfo->intertextsecret && secretexit)
+    if (U_CheckField(gamemapinfo->intertextsecret) && secretexit)
```
Comment (fabiangreffrath) [inline Source/g_game.c:2435]: Is this semantically equivalent? The old code did `return` if `gamemapinfo->intertextsecret` was a valid pointer, even if `gamemapinfo->intertextsecret[0]` was `'-'`. The new code only returns if both conditions are met.

Comment (rfomin) [reply Source/g_game.c]: You right, fixed in https://github.com/fabiangreffrath/woof/pull/550/commits/e31abd09cd2bd68e0e18108c0bd8e5ece1345bde

**Pull request merged**